pkg/config/derived-role-cert.go

```diff
@@ -43,29 +43,44 @@ func (idCfg *IdentityConfig) derivedRoleCertConfig() error {
 		return nil // disabled
 	}
 
-	if idCfg.roleCertDir == "" {
+	if idCfg.roleCertDir == "" && idCfg.roleCertNamingFormat == "" {
 		return nil // disabled
 	}
+	// If both the RoleCert settings and the NamingFormat settings are configured redundantly, an error will be returned.
+	if idCfg.roleCertDir != "" && idCfg.roleCertNamingFormat != "" {
+		return fmt.Errorf("RoleCertDir and RoleCertNamingFormat are both set: RoleCertDir %s, RoleCertNamingFormat %s", idCfg.roleCertDir, idCfg.roleCertNamingFormat)
+	}
+	if idCfg.roleCertDir != "" && idCfg.roleCertKeyNamingFormat != "" {
+		return fmt.Errorf("RoleCertDir and RoleCertKeyNamingFormat are both set: RoleCertDir %s, RoleCertKeyNamingFormat %s", idCfg.roleCertDir, idCfg.roleCertKeyNamingFormat)
+	}
+	// If RoleCertKeyFileOutput is enabled, RoleCertKeyNamingFormat or RoleCertDir must be set.
+	if idCfg.roleCertKeyFileOutput && idCfg.roleCertKeyNamingFormat == "" && idCfg.roleCertDir == "" {
+		return fmt.Errorf("RoleCertKeyFileOutput is enabled but RoleCertKeyNamingFormat and RoleCertDir are not set")
+	}
 
 	// Enabled from now on:
-	dir := strings.TrimSuffix(idCfg.roleCertDir, "/") + "/" // making sure it always ends with `/`
+	var format, keyFormat string
+	if idCfg.roleCertDir == "" {
+		format = idCfg.roleCertNamingFormat
+		keyFormat = idCfg.roleCertKeyNamingFormat
+	} else {
+		// If only RoleCertDir is defined, fixed values will be assigned to format and keyFormat:
+		format = filepath.Join(idCfg.roleCertDir, "{{domain}}{{delimiter}}{{role}}"+".cert.pem")
+		if idCfg.roleCertKeyFileOutput {
+			keyFormat = filepath.Join(idCfg.roleCertDir, "{{domain}}{{delimiter}}{{role}}"+".key.pem")
+		}
+	}
 	idCfg.RoleCert = DerivedRoleCert{
 		Use:               true,
-		Dir:               dir,
 		TargetDomainRoles: idCfg.targetDomainRoles.roleCerts,
-		Format:            dir + "{{domain}}{{delimiter}}{{role}}.cert.pem",
-		KeyFormat: func() string {
-			if idCfg.roleCertKeyFileOutput {
-				return dir + "{{domain}}{{delimiter}}{{role}}.key.pem"
-			}
-			return "" // means no separate key file output feature enabled
-		}(),
-		Delimiter: idCfg.roleCertFilenameDelimiter,
+		Format:            format,
+		KeyFormat:         keyFormat,
+		Delimiter:         idCfg.roleCertFilenameDelimiter,
 	}
```

Comment:
Can we use anonymous functions here for possibly better readability?
just a sample here:
```go
                Format: func() string {
			if idCfg.roleCertDir == "" {
				return idCfg.roleCertNamingFormat
			}
			return filepath.Join(idCfg.roleCertDir, "{{domain}}{{delimiter}}{{role}}.cert.pem")
		}(),
		KeyFormat: func() string {
			if idCfg.roleCertDir == "" {
				return idCfg.roleCertKeyNamingFormat
			}
			// If only RoleCertDir is defined, fixed values will be assigned to format and keyFormat:
			if idCfg.roleCertKeyFileOutput {
				return filepath.Join(idCfg.roleCertDir, "{{domain}}{{delimiter}}{{role}}"+".key.pem")
			}
			return "" // means no separate key file output feature enabled
		}(),
```

Reply:
handled in: https://github.com/AthenZ/k8s-athenz-sia/pull/149/commits/8515d06e7f4342ddf4c18ab94b97774fa107e598